Source/DafnyRuntime/DafnyRuntimeRust/src/lib.rs

```diff
@@ -19,6 +15,10 @@ pub use itertools;
 // An atomic box is just a RefCell in Rust
 pub type SizeT = usize;
 
+// Dafny types must be clonable in constant time
```

Comment:
It is one of my strong preferences for now, although I'll make it configurable in case someone has an extremely large datatype but they still want the datatype itself to be cloned instead of a reference counting type.

The only mutable type in Dafny is a reference, and references can be cloned at no cost, in constant time. The Rust implementation will ensure that.

Reply:
Is it possible to write down what you're using clone() for in general?

Reply:
I wonder if it would worth using `Dupe` (https://developers.facebook.com/blog/post/2021/07/06/rust-nibbles-gazebo-dupe/) here to ensure there are no accidental expensive clones.

Reply:
Yeah, if we were maintaining the generated code, I would definitely consider using a Dupe-like trait, thanks for letting me know. However, since we have full control over the generation pipeline, I agree it's best to reduce the number of dependencies, and clone() is fine for that.